sys/vane/clay.hoon

```diff
@@ -3635,44 +3776,60 @@
 ::
 ++  load
   =>  |%
-      +=  rove-2
-        $%  {$sing p/mood}
-            {$next p/mood q/(unit (each cage lobe))}
-            {$many p/? q/moat r/(map path lobe)}
+      +=  wove-3  rove
+      ++  cult-3  (jug wove-3 duct)
+      ++  dojo-3
+        $:  qyx/cult-3
+            dom/dome
+            dok/(unit dork)
+            mer/(unit mery)
         ==
-      ++  cult-2  (jug rove-2 duct)
-      ++  dojo-2  (cork dojo |=(a/dojo a(qyx *cult-2)))
-      ++  rede-2  (cork rede |=(a/rede a(qyx *cult-2)))
-      ++  room-2  (cork room |=(a/room a(dos (~(run by dos.a) dojo-2))))
-      ++  rung-2  (cork rung |=(a/rung a(rus (~(run by rus.a) rede-2))))
-      ++  raft-2
-        %+  cork  raft
-        |=(a/raft a(fat (~(run by fat.a) room-2), hoy (~(run by hoy.a) rung-2)))
-      ++  axle    $%({$2 ruf/raft-2} {$3 ruf/raft})
+      ++  rede-3
+        $:  lim/@da
+            ref/(unit rind)
+            qyx/cult-3
+            dom/dome
+            dok/(unit dork)
+            mer/(unit mery)
+        ==
+      ++  room-3  (cork room |=(a/room a(dos (~(run by dos.a) dojo-3))))
+      ++  rung-3  (cork rung |=(a/rung a(rus (~(run by rus.a) rede-3))))
+      ++  raft-3
+        $:  fat/(map ship room-3)
+            hoy/(map ship rung-3)
+            ran/rang
+            mon/(map term beam)
+            hez/(unit duct)
+        ==
+      ++  axle    $%({$3 ruf/raft-3} {$4 ruf/raft})
       --
   |=  old/axle
   ^+  ..^$
   ?-  -.old
-    $3  ..^$(ruf ruf.old)
-    $2  =/  rov
-          |=  a/rove-2  ^-  rove
-          ?+  -.a  a
-              $next
-            ?~  q.a  a
-            a(q `q.a)
-          ==
+    $4  ..^$(ruf ruf.old)
+    $3  =/  wov
+          |=  a/wove-3  ^-  wove
+          [~ a]
         =/  cul
-          |=  a/cult-2  ^-  cult
+          |=  a/cult-3  ^-  cult
```

Comment:
I think this would be cleaner as a `|^` since it has so many internal functions.

Reply:
Fair, on it!